examples/reshare/src/orchestrator/actor.rs

```diff
@@ -192,33 +215,60 @@ where
                         // The sender is operating in a newer epoch or the current, ignore. We cannot block them,
```

Comment:
We need to change `engines` to a BTreeMap (not only is the current code non-deterministic the logic to get latest is broken).